arangod/RestHandler/RestBaseHandler.cpp

```diff
@@ -48,7 +48,11 @@ RestBaseHandler::RestBaseHandler(ArangodServer& server, GeneralRequest* request,
 arangodb::velocypack::Slice RestBaseHandler::parseVPackBody(bool& success) {
   try {
     success = true;
-    return _request->payload(true);
+    auto slice = _request->payload(true);
+    if (_request->contentType() != rest::ContentType::VPACK) {
```

Comment:
nit: if we are calling `_request->payload(...)` multiple times for the same request, then we will account the memory multiple times here. This is not ideal, because `_request->payload(...)` will allocate memory only for the first time if the Content-Type is application/json, but not for subsequent calls.